gen/templates/integration_resource.go.tmpl

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	mondoov1 "go.mondoo.com/mondoo-go"
+)
+
+var _ resource.Resource = (*integration{{.ResourceClassName}}Resource)(nil)
+
+func NewIntegration{{.ResourceClassName}}Resource() resource.Resource {
+	return &integration{{.ResourceClassName}}Resource{}
+}
+
+type integration{{.ResourceClassName}}Resource struct {
+	client *ExtendedGqlClient
+}
+
+type integration{{.ResourceClassName}}ResourceModel struct {
+	// scope
+	SpaceID types.String `tfsdk:"space_id"`
+
+	// integration details
+	Mrn  types.String `tfsdk:"mrn"`
+	Name types.String `tfsdk:"name"`
+
+	// {{.ResourceClassName}} options
+}
+
+func (m integration{{.ResourceClassName}}ResourceModel) GetConfigurationOptions() *mondoov1.{{.ResourceClassName}}ConfigurationOptionsInput {
```

Comment:
For best practices, we should also add generated comments here so that its easier to understand what the methods are doing

Reply:
I think this comment is still valid